hdc/algo/season.py

```diff
@@ -0,0 +1,214 @@
+                seasons.
+        """
+        self.season_range = season_range
+        self.validate_season_ranges()
+
+        if isinstance(date, Dekad):
+            self._seas = self.season_label(date.start_date)
+        elif isinstance(date, str):
+            dekad_date = Dekad(date)
+            self._seas = self.season_label(dekad_date.start_date)
+        elif isinstance(date, int):
+            self._seas = date
+        elif isinstance(date, (datetime, date)):
+            self._seas = self.season_label(date)
+        else:
+            raise ValueError(
+                "Invalid date format. Must be a string, datetime, date, or Dekad."
+            )
+
+    def __repr__(self):
+        """Return string representation."""
+        return f"Season(season_range={self.season_range}, id={self._seas})"
+
+    def __hash__(self):
+        """Return Hash."""
+        return hash(self._seas)
+
+    def __eq__(self, other):
+        """Check equality with other Season or int."""
+        if not isinstance(other, Season):
+            return False
+        return self._seas == other._seas
+
+    def __lt__(self, other):
+        """Check for less than inequality with other Season or int."""
+        if not isinstance(other, Season):
+            return NotImplemented
+        return self._seas < other._seas
+
+    def __le__(self, other):
+        """Check for less than or equal inequality with other Season or int."""
+        if not isinstance(other, Season):
+            return NotImplemented
+        return self._seas <= other._seas
+
+    def __gt__(self, other):
+        """Check for greater than inequality with other Season or int."""
+        if not isinstance(other, Season):
+            return NotImplemented
+        return self._seas > other._seas
+
+    def __ge__(self, other):
+        """Check for greater than or equal inequality with other Season or int."""
+        if not isinstance(other, Season):
+            return NotImplemented
+        return self._seas >= other._seas
+
+    @property
+    def id(self) -> int:
+        """Returns the unique string representation of the season."""
+        return self._seas
+
+    def season_index(self, dekad_of_year: int) -> Optional[int]:
+        """
+        Return the season index (e.g., 1, 2, etc.) for the given dekad of the year.
+
+        Args:
+            dekad_of_year (int): Dekad index (1-36).
+
+        Returns:
+            int: Season index or None if no match.
+        """
+        for i, (start, end) in enumerate(self.season_range):
+            if start <= end:  # Normal case
+                if start <= dekad_of_year <= end:
+                    return i + 1
+            else:  # Cross-year case
+                if dekad_of_year >= start or dekad_of_year <= end:
+                    return i + 1
+        return None
+
+    def season_label(self, date: Union[datetime]) -> int:
```

Comment:
This method could also support all the input types Union[str, int, datetime, date, Dekad]